### PR TITLE
Created app-accordion items

### LINKED ISSUE #123: Create AppAccordion

### PR DESCRIPTION
**Pull request by D1nviz**
- Created app-accordion
- Used reexport file, because this components always are using together 
- Added tests, styles

https://github.com/idx-academy/spa-orders/assets/118522791/4bf1dd93-3fc4-41d7-9cbc-9339385b103f

### CODE REVIEW

**Review by heshvab — APPROVED** [review]

File: code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@import "@design-system";
+
+.spa-accordion-container {
+  &--expanded {
+    margin: 0 !important; //necessary for overriding MUI styles
+  }
+  box-shadow: none;
+  border: 1px solid $ghost-alpha;
```
Comment (dmtrth25) [inline code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss]: can use our `box-shadow` from design system. can add your property and reuse it
also add variable for border color with `var` like our previous examples of usage

File: code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+describe("AppAccordionContainer", () => {
+  test("renders AppAccordionContainer expanded", () => {
+    render(
+      <AppAccordionContainer expanded>
+        <div>Child</div>
+      </AppAccordionContainer>
+    );
+    const accordionElement = screen
+      .getByText("Child")
+      .closest(".spa-accordion-container");
```
Comment (dmtrth25) [inline code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.test.tsx]: Is it necessary here ? Can the test be passed without `closest` ?

File: code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss (hunk 1 above)
Comment (D1nviz) [reply code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss]: fixed

File: code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.test.tsx (hunk 2 above)
Comment (D1nviz) [reply code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.test.tsx]: replaced by `parentElement` property

File: code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss (hunk 1 above)
Comment (dmtrth25) [reply code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss]: > fixed

remove usage of `box-shadow` or implement it via design system and also forget to remove `,` for `$ghost-alpha` at the end

We already have properties for that just add own
image

Comment (D1nviz) [reply code/src/components/app-accordion/app-accorion-container/AppAccordionContainer.scss]: Added

**Review by dmtrth25 — APPROVED** [review]

File: code/src/components/app-accordion/app-accordion-details/AppAccordionDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import AccordionDetails from "@mui/material/AccordionDetails";
+import { AppAccordionDetailsProps } from "@/components/app-accordion/app-accordion-details/AppAccordionDetails.types";
+
+const AppAccordionDetails = ({ ...props }: AppAccordionDetailsProps) => {
```
Comment (mhevyk) [inline code/src/components/app-accordion/app-accordion-details/AppAccordionDetails.tsx]: We don't need descructuring here

File: code/src/components/app-accordion/app-accordion-summary/AppAccordionSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import AccordionSummary from "@mui/material/AccordionSummary";
+import { AppAccordionSummaryProps } from "@/components/app-accordion/app-accordion-summary/AppAccordionSummary.types";
+
+import "@/components/app-accordion/app-accorion-container/AppAccordionContainer.scss";
+
+const AppAccordionSummary = ({ ...props }: AppAccordionSummaryProps) => {
```
Comment (mhevyk) [inline code/src/components/app-accordion/app-accordion-summary/AppAccordionSummary.tsx]: Same here

File: code/src/components/app-accordion/app-accordion-details/AppAccordionDetails.tsx (hunk 3 above)
Comment (D1nviz) [reply code/src/components/app-accordion/app-accordion-details/AppAccordionDetails.tsx]: fixed

File: code/src/components/app-accordion/app-accordion-summary/AppAccordionSummary.tsx (hunk 4 above)
Comment (D1nviz) [reply code/src/components/app-accordion/app-accordion-summary/AppAccordionSummary.tsx]: fixed

**Review by mhevyk — APPROVED** [review]

**Pull request merged**